From 8b2442a927d0141dbf06289d20486e0e631ebec7 Mon Sep 17 00:00:00 2001
From: maximilian attems <max@stro.at>
Date: Fri, 5 Sep 2008 22:13:21 +0200
Subject: [PATCH 06/21] [klibc] mount: list fs of particular type

makes possible: mount -t squashfs
pass type of the fs down to print_mount().

fixes http://bugs.debian.org/491067
requested for casper live cd initramfs.

Cc: Colin Watson <cjwatson@ubuntu.com>
Signed-off-by: maximilian attems <max@stro.at>
---
 usr/utils/mount_main.c |   10 ++++++----
 1 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/usr/utils/mount_main.c b/usr/utils/mount_main.c
index 2616e46..89e75d7 100644
--- a/usr/utils/mount_main.c
+++ b/usr/utils/mount_main.c
@@ -26,7 +26,7 @@ static __noreturn usage(void)
 	exit(1);
 }
 
-static __noreturn print_mount(void)
+static __noreturn print_mount(char *type)
 {
 	FILE *mfp;
 	struct mntent *mnt;
@@ -40,6 +40,8 @@ static __noreturn print_mount(void)
 	while ((mnt = getmntent(mfp)) != NULL) {
 		if (mnt->mnt_fsname && !strncmp(mnt->mnt_fsname, "no", 2))
 			continue;
+		if (type && mnt->mnt_type && strcmp(type, mnt->mnt_type))
+			continue;
 		printf("%s on %s", mnt->mnt_fsname, mnt->mnt_dir);
 		if (mnt->mnt_type != NULL && mnt->mnt_type != '\0')
 			printf (" type %s", mnt->mnt_type);
@@ -136,6 +138,9 @@ int main(int argc, char *argv[])
 		}
 	} while (1);
 
+	if (optind == argc)
+		print_mount(type);
+
 	/*
 	 * If remount, bind or move was specified, then we don't
 	 * have a "type" as such.  Use the dummy "none" type.
@@ -143,9 +148,6 @@ int main(int argc, char *argv[])
 	if (rwflag & MS_TYPE)
 		type = "none";
 
-	if (optind == argc)
-		print_mount();
-
 	if (optind + 2 != argc || type == NULL)
 		usage();
 
-- 
1.6.5

